src/client/datascience/jupyter/kernels/kernel.ts

```diff
@@ -231,10 +221,19 @@ export class Kernel implements IKernel {
                     }
                     resolve(this.notebook);
                 } catch (ex) {
+                    sendKernelTelemetryEvent(
```

Comment:
Moved this here, as we need to handle all errors, previously we were only interested in creation Of notebooks.